### PR TITLE
Enable Transport configuration for git repos in a composite

### PR DESCRIPTION
**Pull request by pivotal-dylan-roberts**
Merge Ssh configuration properties classes with JGitEnvironmentProperties

Fixes gh-976

### CODE REVIEW

**Comment by pivotal-dylan-roberts** [comment]
One more change to make before this is ready for review!

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/MultipleJGitEnvironmentRepositoryFactory.java [hunk 1]
```diff
@@ -29,23 +29,27 @@
...
 		this.server = server;
-		this.transportConfigCallback = transportConfigCallback;
 	}
 
 	@Override
 	public MultipleJGitEnvironmentRepository build(MultipleJGitEnvironmentProperties environmentProperties) {
 		MultipleJGitEnvironmentRepository repository = new MultipleJGitEnvironmentRepository(environment,
 				environmentProperties);
-		repository.setTransportConfigCallback(transportConfigCallback.orElse(null));
+		repository.setTransportConfigCallback(propertiesBasedSshTransportCallback(environmentProperties));
```
Comment (tdanylchuk) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/MultipleJGitEnvironmentRepositoryFactory.java]: I guess it's better to leave `transportConfigCallback` for custom configuration and check whether it's provided or not here.

**Comment by pivotal-dylan-roberts** [comment]
@tdanylchuk Updated per your suggestion!

**Comment by ojhughes** [comment]
LGTM 👍

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/ssh/PropertiesBasedSshTransportConfigCallback.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package org.springframework.cloud.config.server.ssh;
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/ssh/PropertiesBasedSshTransportConfigCallback.java]: You need a copyright header here

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java [hunk 3]
```diff
@@ -114,14 +113,14 @@ public EnvironmentWatch environmentWatch() {
 	private ConfigServerProperties server;
 
 	@Autowired(required = false)
-	private TransportConfigCallback transportConfigCallback;
+    private TransportConfigCallback customTransportConfigCallback;
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java]: spacing looks off here

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/ssh/FileBasedSshTransportConfigCallback.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package org.springframework.cloud.config.server.ssh;
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/ssh/FileBasedSshTransportConfigCallback.java]: need the copyright header

Comment (pivotal-dylan-roberts) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/ssh/FileBasedSshTransportConfigCallback.java]: Done!

**Comment by pivotal-dylan-roberts** [comment]
@spencergibb @ryanjbaxter Requested changes implemented!

**Pull request merged**